### PR TITLE
ipscan: init at 3.6.2

### PR DESCRIPTION
**Pull request by kylesferrazza**
###### Motivation for this change
Wanted to use `ipscan`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by kylesferrazza** [comment]
Will add desktop file.

File: pkgs/tools/security/ipscan/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchurl, jdk, jre, swt, makeWrapper, xorg }:
+
+let
+  pname = "ipscan";
+  version = "3.6.2";
+
+  jar = fetchurl {
```
Comment (flokli) [inline pkgs/tools/security/ipscan/default.nix]:
```suggestion
  src = fetchurl {
```

File: pkgs/tools/security/ipscan/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      --add-flags "-Xmx256m -Djava.library.path=${swt}/lib -cp ${jar}:${swt}/jars/swt.jar net.azib.ipscan.Main"
+  '';
+
+  passthru.jar = jar;
+
+  meta = with stdenv.lib; {
+    description = "Fast and friendly network scanner";
+    homepage = https://angryip.org;
+    license = licenses.gpl2;
+    platforms = platforms.linux;
```
Comment (flokli) [inline pkgs/tools/security/ipscan/default.nix]: `linux64` in the source suggests this is `x86_64-linux` only.

File: pkgs/tools/security/ipscan/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  installPhase = ''
+    mkdir -p $out/share/java
+    ln -s ${jar} ${jarPath}
+    makeWrapper ${jre}/bin/java $out/bin/ipscan \
+      --prefix LD_LIBRARY_PATH : "$out/lib/:${stdenv.lib.makeLibraryPath [ swt xorg.libXtst ]}" \
+      --add-flags "-Xmx256m -Djava.library.path=${swt}/lib -cp ${jar}:${swt}/jars/swt.jar net.azib.ipscan.Main"
+  '';
+
+  passthru.jar = jar;
```
Comment (flokli) [inline pkgs/tools/security/ipscan/default.nix]: This can be dropped, or for what do you need it?

File: pkgs/tools/security/ipscan/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  pname = "ipscan";
+  version = "3.6.2";
+
+  jar = fetchurl {
+    url = "https://github.com/angryip/ipscan/releases/download/${version}/ipscan-linux64-${version}.jar";
+    sha256 = "0fmrrxanw46g53jvlb50516jqwqjzky564bf3df9i3ymjffrfnw4";
+  };
+
+  jarName = "${pname}-${version}.jar";
+  jarPath = "$out/share/java/${jarName}";
```
Comment (flokli) [inline pkgs/tools/security/ipscan/default.nix]: please inline these into the `installPhase`, you're not using them elsewhere.

File: pkgs/tools/security/ipscan/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+in stdenv.mkDerivation rec {
+  inherit pname version;
+
+  dontUnpack = true;
+
+  buildInputs = [ makeWrapper jdk ];
+
+  installPhase = ''
+    mkdir -p $out/share/java
+    ln -s ${jar} ${jarPath}
```
Comment (flokli) [inline pkgs/tools/security/ipscan/default.nix]:
```suggestion
    cp $src $out/share/${pname}.jar
```

as per `doc/languages-frameworks/java.xml`, this is not something imported by other java packages, right?

File: pkgs/tools/security/ipscan/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    makeWrapper ${jre}/bin/java $out/bin/ipscan \
+      --prefix LD_LIBRARY_PATH : "$out/lib/:${stdenv.lib.makeLibraryPath [ swt xorg.libXtst ]}" \
+      --add-flags "-Xmx256m -Djava.library.path=${swt}/lib -cp ${jar}:${swt}/jars/swt.jar net.azib.ipscan.Main"
```
Comment (flokli) [inline pkgs/tools/security/ipscan/default.nix]:
```suggestion
      --add-flags "-Xmx256m -cp $out/share/${pname}.jar:${swt}/jars/swt.jar net.azib.ipscan.Main
```

**Review by flokli — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/ipscan/default.nix (hunk 2 above)
Comment (kylesferrazza) [reply pkgs/tools/security/ipscan/default.nix]: Correct, I haven't tested on macOS, and seems like there is a separate jar for x32 that I haven't tested yet either.

Comment (kylesferrazza) [reply pkgs/tools/security/ipscan/default.nix]: Ah, I see you what you meant now (restrict the platforms further), done.

**Comment by kylesferrazza** [comment]
@GrahamcOfBorg build ipscan

**Pull request merged**